### ISSUE TITLE
Evaluation result far from training's evaluation result

### ISSUE THREAD

**Issue by wenwen12321** [Vegetebird/MHFormer#58]
Hi, author~

I have a weird problem: sometimes, my evaluation result is far from my training result. 
I cannot understand what's going on.
Only modified parameter setting: **batch_size=128**, **frames=9**.

Following is my training log:

> 2022/06/01 16:42:57 epoch: 1, lr: 0.0010000, loss: 0.0743, p1: 61.72, p2: 44.22
> 2022/06/01 17:37:46 epoch: 2, lr: 0.0009500, loss: 0.0515, p1: 55.41, p2: 42.16
2022/06/01 18:32:33 epoch: 3, lr: 0.0009025, loss: 0.0475, p1: 52.42, p2: 40.21
2022/06/01 19:27:29 epoch: 4, lr: 0.0008574, loss: 0.0453, p1: 56.34, p2: 41.44
2022/06/01 20:22:25 epoch: 5, lr: 0.0008145, loss: 0.0438, p1: 52.88, p2: 40.48
2022/06/01 21:17:29 epoch: 6, lr: 0.0004073, loss: 0.0418, p1: 52.69, p2: 39.66
2022/06/01 22:12:27 epoch: 7, lr: 0.0003869, loss: 0.0412, p1: 52.71, p2: 40.04
2022/06/01 23:07:21 epoch: 8, lr: 0.0003675, loss: 0.0409, p1: 54.99, p2: 40.65
2022/06/02 00:02:15 epoch: 9, lr: 0.0003492, loss: 0.0404, p1: 51.59, p2: 40.14
2022/06/02 00:57:07 epoch: 10, lr: 0.0003317, loss: 0.0401, p1: 51.22, p2: 39.64
2022/06/02 01:52:05 epoch: 11, lr: 0.0001659, loss: 0.0395, p1: 53.70, p2: 40.66
2022/06/02 02:46:58 epoch: 12, lr: 0.0001576, loss: 0.0392, p1: 53.23, p2: 40.64
2022/06/02 03:41:58 epoch: 13, lr: 0.0001497, loss: 0.0391, p1: 52.64, p2: 40.32
2022/06/02 04:36:47 epoch: 14, lr: 0.0001422, loss: 0.0390, p1: 50.99, p2: 39.53

and the following is my testing result:

> ===Action=== ==p#1 mm== =p#2 mm=
Directions   786.19     441.17
Discussion   765.48     417.01
Eating       772.33     402.92
Greeting     792.56     444.53
Phoning      762.85     388.00
Photo        808.33     432.77
Posing       786.53     437.86
Purchases    772.91     409.88
Sitting      748.61     378.83
SittingDown  732.58     360.98
Smoking      752.08     388.04
Waiting      771.78     414.62
WalkDog      773.06     403.30
Walking      774.60     407.94
WalkTogether 788.86     436.25
Average      772.58     410.94
p1: 772.58, p2: 410.94

Does it mean overfitting? But I didn't modify the paper's code except for batch_size and frames number.

**Comment by Vegetebird** [comment]
The bug have been fixed a few weeks ago. You can redownload the repo and pretrained models, and retrain the model.

**Comment by wenwen12321** [comment]
Got it! thank you so much.
I will redownload your repo and try it again.

I still curious what bug that would cause this issue??
Actually, I am facing the same issue in other paper's code, but still don't know how to fix it.

**Comment by Vegetebird** [comment]
Maybe the bug occurs when loading the pretrained model (https://github.com/Vegetebird/MHFormer/blob/e48271cc6e7e92c13560295db6ce334a974d8ad9/main.py#L123). 

The module name in pretrained model (pre_dict .keys()) and defined model (model_dict.keys()) are different, so it does not load some weights of modules, leading to incorrect results.

**Comment by wenwen12321** [comment]
Hi, thanks for your prompt reply.

I have redownloaded your repo and retrained again.
The first time I train can get the normal evaluation result.
But when I train with modified a little bit of your model structure, it still got the following result.

> 2022/06/28 03:38:43 epoch: 1, lr: 0.0010000, loss: 0.0811, p1: 54.36, p2: 42.61
2022/06/28 03:58:35 epoch: 2, lr: 0.0009500, loss: 0.0468, p1: 53.80, p2: 40.96
2022/06/28 04:18:27 epoch: 3, lr: 0.0009025, loss: 0.0414, p1: 51.79, p2: 40.03
2022/06/28 04:38:18 epoch: 4, lr: 0.0008574, loss: 0.0385, p1: 50.77, p2: 39.43
2022/06/28 04:58:10 epoch: 5, lr: 0.0008145, loss: 0.0366, p1: 50.59, p2: 39.05
2022/06/28 05:18:02 epoch: 6, lr: 0.0004073, loss: 0.0338, p1: 49.91, p2: 38.41
2022/06/28 05:37:54 epoch: 7, lr: 0.0003869, loss: 0.0331, p1: 49.47, p2: 38.23
2022/06/28 05:57:45 epoch: 8, lr: 0.0003675, loss: 0.0326, p1: 49.49, p2: 38.25
2022/06/28 06:17:37 epoch: 9, lr: 0.0003492, loss: 0.0321, p1: 49.76, p2: 38.30
2022/06/28 06:37:28 epoch: 10, lr: 0.0003317, loss: 0.0317, p1: 50.06, p2: 38.16
2022/06/28 06:57:20 epoch: 11, lr: 0.0001659, loss: 0.0307, p1: 49.28, p2: 38.06
2022/06/28 07:17:12 epoch: 12, lr: 0.0001576, loss: 0.0305, p1: 50.01, p2: 38.46
2022/06/28 07:37:04 epoch: 13, lr: 0.0001497, loss: 0.0303, p1: 49.58, p2: 38.27
2022/06/28 07:56:56 epoch: 14, lr: 0.0001422, loss: 0.0301, p1: 50.25, p2: 38.50

===Action=== ==p#1 mm== =p#2 mm=
Directions   469.51     208.62
Discussion   441.79     220.50
Eating       401.69     216.48
Greeting     463.40     226.62
Phoning      382.94     218.79
Photo        446.51     230.31
Posing       460.14     218.05
Purchases    489.61     220.97
Sitting      383.07     223.03
SittingDown  385.89     238.07
Smoking      387.83     221.27
Waiting      444.74     211.86
WalkDog      441.39     225.07
Walking      385.67     231.80
WalkTogether 410.36     237.21
Average      426.30     223.24
p1: 426.30, p2: 223.24

**Comment by wenwen12321** [comment]
Thank you so much!
It solved my problem.
